gobblin-core/src/main/java/gobblin/metrics/GobblinMetrics.java

```diff
@@ -548,6 +557,155 @@ private void buildKafkaMetricReporter(Properties properties) {
     LOGGER.info("Will start reporting metrics to Kafka");
   }
 
+  private void buildGraphiteMetricReporter(Properties properties) {
+    boolean metricsEnabled =
+        Utils.getPropAsBoolean(properties, ConfigurationKeys.METRICS_REPORTING_GRAPHITE_METRICS_ENABLED_KEY,
+            ConfigurationKeys.DEFAULT_METRICS_REPORTING_GRAPHITE_METRICS_ENABLED);
+    boolean eventsEnabled =
+        Utils.getPropAsBoolean(properties, ConfigurationKeys.METRICS_REPORTING_GRAPHITE_EVENTS_ENABLED_KEY,
+            ConfigurationKeys.DEFAULT_METRICS_REPORTING_GRAPHITE_EVENTS_ENABLED);
+    if (!metricsEnabled && !eventsEnabled) {
+      LOGGER.info("Not reporting metrics and events to Graphite");
+      return;
+    }
+    
+    try {
+      Preconditions.checkArgument(properties.containsKey(ConfigurationKeys.METRICS_REPORTING_GRAPHITE_HOSTNAME),
```

Comment:
I followed the pattern of the other reporters: catch exception, log it and return. Should any issue in the metric reporting bring down the job?